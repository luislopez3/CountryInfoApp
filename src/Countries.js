import React, { useEffect, useState } from "react";
import { Link, Route, useParams, useRouteMatch } from "react-router-dom";
import ShowDetails from "./ShowDetails";
import { allCountries, countryByRegion } from "./api-requests";

export default function Countries() {
  const [countries, setCountries] = useState([]);
  const { region } = useParams();
  const { url, path } = useRouteMatch();

  useEffect(() => {
    if (region) {
      countryByRegion(region).then((response) => {
        setCountries(response);
      });
    } else {
      allCountries().then((response) => {
        setCountries(response);
      });
    }
  }, [region]);

  return (
    <div>
      <h1>Countries</h1>
      <p>Select a Country to view the details about it.</p>
      {countries.map((country, index) => {
        return (
          <div className="country-row">
            <div className="country-column">
              <ul key={index}>
                <Link to={`${url}/${country.name}`}>
                  <button>{country.name}</button>
                </Link>
              </ul>
            </div>
          </div>
        );
      })}
      <div>
        <hr />
        <Route path={`${path}/:country`}>
          <ShowDetails />
        </Route>
      </div>
    </div>
  );
}
