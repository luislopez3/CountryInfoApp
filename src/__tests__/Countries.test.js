import React from 'react';
import { create } from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import Countries from '../Countries.js';

describe("Show Details component", () => {
    test("Matches the snapshot", () => {
        const wrapper = create(
        <MemoryRouter ><Countries /></MemoryRouter>
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});


