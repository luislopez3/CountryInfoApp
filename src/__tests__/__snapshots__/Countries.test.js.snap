// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Details component Matches the snapshot 1`] = `
<div>
  <h2>
    Countries
  </h2>
  <p>
    Select a Country to view the details about it.
  </p>
  <div>
    <hr />
  </div>
</div>
`;
